netbox/utilities/versions.py

```diff
@@ -0,0 +1,35 @@
+import requests
+from cacheops import cached
+from django.conf import settings
+from packaging import version
+
+
+@cached(timeout=settings.GITHUB_VERSION_TIMEOUT if settings.GITHUB_VERSION_TIMEOUT > 0 else 1)
```

Comment:
It doesn't seem prudent to default to a 1 second timeout here, as that would result in a new check for virtually every page load. This isn't a great place for validation, anyway: `GITHUB_VERSION_TIMEOUT` should be validated immediately upon import in `settings.py`, so that any code which references it can be sure to receive a valid value.

Reply:
hSaria actually posted a great example of how to work around this in a test here: https://github.com/netbox-community/netbox/issues/3952#issuecomment-575777381